specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-03-01/virtualMachineScaleSet.json

```diff
@@ -3948,6 +3948,61 @@
       },
       "description": "The configuration parameters used while performing resilient VM deletion."
     },
+    "ScheduledEventsPolicy": {
+      "type": "object",
+      "description": "Specifies Redeploy,Reboot and scheduledEventsAdditionalPublishingTargets Scheduled Event related configurations.",
+      "properties": {
+        "UserInitiatedRedeploy": {
+          "$ref": "#/definitions/UserInitiatedRedeploy",
+          "description": "The configuration parameters used while creating userInitiatedRedeploy scheduled event setting creation."
+        },
+        "UserInitiatedReboot": {
+          "$ref": "#/definitions/UserInitiatedReboot",
+          "description": "The configuration parameters used while creating userInitiatedReboot scheduled event setting creation."
+        },
+        "ScheduledEventsAdditionalPublishingTargets": {
+          "$ref": "#/definitions/ScheduledEventsAdditionalPublishingTargets",
+          "description": "The configuration parameters used while publishing scheduledEventsAdditionalPublishingTargets."
+        }
+      }
+    },
+    "UserInitiatedRedeploy": {
+      "type": "object",
+      "description": "Specifies Redeploy related Scheduled Event related configurations.",
+      "properties": {
+        "automaticallyApprove": {
+          "type": "boolean",
+          "description": "Specifies Redeploy Scheduled Event related configurations."
```

Comment:
ARM recommends enums over booleans for better extensibility: 

[https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md)


FOr example hre , you ocould have 

"ApprovalMode": "Automatic", allowing for more options for approval modes later. this also prevents bloat of boooleans in the payload